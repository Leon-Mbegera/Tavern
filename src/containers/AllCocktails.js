import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const AllCocktails = ({ cocktails }) => (
  <div className="container-fluid row wrapper">
    {cocktails.map((cocktail, idx) => (
      <>
        <div className="col-md-3">
          <div className="card bg-warning mb-3" style={{ width: '20rem' }}>
            <img className="card-img-top" src={cocktail.strDrinkThumb} alt={cocktail.strDrinkThumb} />
            <div className="card-body">
              <h6 className="card-title fw-bolder" key={cocktail.idDrink}>{cocktail.strDrink}</h6>
              <Link className="btn btn-danger btn-block" to={`/cocktail/${idx}`}>Cocktail</Link>
            </div>
          </div>
        </div>
      </>
    ))}
  </div>
);

AllCocktails.propTypes = {
  cocktails: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default AllCocktails;
