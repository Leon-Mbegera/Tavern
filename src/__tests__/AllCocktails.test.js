import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import 'regenerator-runtime';
import AllCocktails from '../containers/AllCocktails';

test('snapshot of the all cocktails page', () => {
  const cocktails = [{ idDrink: '15346' }];
  const Home = renderer.create(
    <BrowserRouter>
      <AllCocktails cocktails={cocktails} />
    </BrowserRouter>,
  ).toJSON();
  expect(Home).toMatchSnapshot();
});

test('Each cocktail card component is rendered', () => {
  const cocktails = [{ idDrink: '32144' }];
  const card = renderer.create(
    <BrowserRouter>
      <AllCocktails cocktails={cocktails} />
    </BrowserRouter>,
  ).toJSON();
  expect(card).toBeDefined();
});

test('classes assigned to the cocktail cards', async () => {
  const cocktails = [{ idDrink: '32144' }];
  const card = render(
    <BrowserRouter>
      <AllCocktails cocktails={cocktails} />
    </BrowserRouter>,
  );
  const cocktailCard = await card.findByTestId('cocktail-card');
  expect(cocktailCard.className).toContain('card bg-warning mb-3');
});
