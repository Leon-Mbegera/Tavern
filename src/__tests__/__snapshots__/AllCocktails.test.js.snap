// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of the all cocktails page 1`] = `
<div
  className="container-fluid row wrapper"
>
  <div
    className="col-md-3"
  >
    <div
      className="card bg-warning mb-3"
      data-testid="cocktail-card"
      style={
        Object {
          "width": "20rem",
        }
      }
    >
      <img
        className="card-img-top"
      />
      <div
        className="card-body"
      >
        <h6
          className="card-title fw-bold"
        />
        <a
          className="btn btn-danger btn-block"
          href="/cocktail/0"
          onClick={[Function]}
        >
          Cocktail
        </a>
      </div>
    </div>
  </div>
</div>
`;
